### ISSUE TITLE
Excess output of the FaRL model wrapper

### ISSUE THREAD

**Issue by detkov** [FacePerceiver/facer#2]
Here is the part of the code from the inference of the FaRL model wrapper:
https://github.com/FacePerceiver/facer/blob/d6535a617c11d33158ac8ae5d8d8af156e2b3e82/facer/face_parsing/farl.py#L71-L76

Why do you propose to return in every inference such information, as `label_names`, while this can be obtained, for example, with `model.label_names` field outside of the `inference`? The same with the `data` itself — you can obtain the `data` later after inference from the same variable you have passed into `inference`. Why not return only the processed output of the model, `seg_logits`? 

Could you please explain the motivation behind it?

**Comment by YANG-H** [comment]
Hi @detkov ,  thanks for sharing your thoughts.

We design this to avoid additional queries of `label_names`. By giving `label_names` as part of the output, we can get the meaning of each logit channel without knowing which face parser is called (we will add more face parsers later). 

For example, our visualization utility `facer.draw_bchw` could paint the output `logits` according to the embedded `label_names` without taking an additional `label_names` as input. 

BTW, does this design cause any problems in any environments?

**Comment by detkov** [comment]
No problems with env at all.
I got your point, but I still think that it is better to separate the responsibilities, return from the `inference` function just the processed output of the network and change depending visualization utilities with respect to it. 

Nevertheless, thanks for the answer and thanks for the provided module.

What are the next face parsing models you are going to deliver? Will there be left and right ear classes?
